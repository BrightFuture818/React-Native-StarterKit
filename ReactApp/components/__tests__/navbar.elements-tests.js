/**
 * Test to check if the component renderes correctly
 */
'use strict';

import 'react-native'
import React from 'react'
import NavbarElements from '../navbar.elements'
import renderer from 'react-test-renderer'

/*
 * Check if alerts renders correcly
 * and asserting it to the matching snapshot
 */
it('Navbar title renders correcly', () => {

  const tree = renderer.create(
    <NavbarElements.Title />
  ).toJSON()

  expect(tree).toMatchSnapshot()

});


/**
 * Check if alerts renders correcly
 * and asserting it to the matching snapshot
 */
it('Navbar left button renders correcly', () => {

  const tree = renderer.create(
    <NavbarElements.LeftButton
       onPress={ e => e}
       icon='ios-menu-outline'
     />
  ).toJSON()

  expect(tree).toMatchSnapshot()

});
