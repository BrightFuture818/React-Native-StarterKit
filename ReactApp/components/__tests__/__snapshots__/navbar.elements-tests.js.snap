exports[`test Navbar left button renders correcly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={
    Object {
      "bottom": 7,
      "left": 7,
      "right": 7,
      "top": 7,
    }
  }
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "left": 20,
      "opacity": 1,
      "top": 4,
    }
  }
  testID={undefined}>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontFamily": "Ionicons",
          "fontSize": 36,
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        undefined,
      ]
    }>
    ï‡‚
  </Text>
</View>
`;

exports[`test Navbar title renders correcly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#555",
        "fontFamily": "Avenir",
        "fontSize": 14,
        "fontWeight": "500",
        "lineHeight": 21,
      },
      Object {
        "fontWeight": "900",
      },
      Object {
        "bottom": 6,
        "color": "#FFFFFF",
        "fontSize": 13,
      },
    ]
  }>
  Starter Kit
</Text>
`;
